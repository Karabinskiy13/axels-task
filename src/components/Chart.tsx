import React, { useEffect, useState } from 'react';

import { useNavigate } from 'react-router-dom';
import { Doughnut, Pie, Line } from 'react-chartjs-2';
import { Button, CircularProgress, Box } from '@mui/material';
import {
  Chart as ChartJS,
  ArcElement,
  Tooltip,
  Legend,
  CategoryScale,
  LinearScale,
  LineElement,
  PointElement,
  Title
} from 'chart.js';

import { pictureService } from '../services/picture.service';
import { Charts, Wrapper } from '../styled/Charts';

ChartJS.register(
  ArcElement,
  Tooltip,
  Legend,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);

const getData = (data: number[]) => {
  return {
    labels: ['Dogs', 'Horses', 'Flowers', 'Mountains', 'Rivers'],
    datasets: [
      {
        label: 'Total by query',
        data,
        borderWidth: 2,
        backgroundColor: [
          'rgba(255, 99, 132, 0.2)',
          'rgba(54, 162, 235, 0.2)',
          'rgba(255, 206, 86, 0.2)',
          'rgba(75, 192, 192, 0.2)',
          'rgba(153, 102, 255, 0.2)'
        ],
        borderColor: [
          'rgba(255, 99, 132, 1)',
          'rgba(54, 162, 235, 1)',
          'rgba(255, 206, 86, 1)',
          'rgba(75, 192, 192, 1)',
          'rgba(153, 102, 255, 1)'
        ]
      }
    ]
  };
};

const Chart = () => {
  const navigate = useNavigate();
  const [data, setData] = useState(getData([]));
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetch = async () => {
      setLoading(true);
      const { total: totalDogs } = await pictureService.getImagesByQuery('dogs', 1);
      const { total: totalHorses } = await pictureService.getImagesByQuery('horses', 1);
      const { total: totalFlowers } = await pictureService.getImagesByQuery('flowers', 1);
      const { total: totalMountains } = await pictureService.getImagesByQuery('mountains', 1);
      const { total: totalRivers } = await pictureService.getImagesByQuery('rivers', 1);

      setData(getData([totalDogs, totalHorses, totalFlowers, totalMountains, totalRivers]));
      setLoading(false);
    };

    fetch();
  }, []);

  return (
    <Box
      height="100vh"
      display="flex"
      justifyContent="center"
      alignItems="center"
      flexDirection="column">
      {!loading && (
        <Wrapper>
          <Button onClick={() => navigate('/')} variant="outlined">
            Back to Pictures
          </Button>
          <Charts>
            <Doughnut data={data} />
          </Charts>
          <Charts>
            <Pie data={data} />
          </Charts>
          <Charts>
            <Line data={data} />
          </Charts>
          <Button
            variant="contained"
            onClick={() =>
              setData(
                getData([
                  Math.floor(Math.random() * 10000),
                  Math.floor(Math.random() * 10000),
                  Math.floor(Math.random() * 10000),
                  Math.floor(Math.random() * 10000),
                  Math.floor(Math.random() * 10000)
                ])
              )
            }>
            Change Data
          </Button>
        </Wrapper>
      )}
      {loading && <CircularProgress size={100} />}
    </Box>
  );
};

export default Chart;
