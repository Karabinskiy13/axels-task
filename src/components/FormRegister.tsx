import React, { useState } from 'react';
import { useNavigate } from 'react-router';
import { Body, Button, Form, Input, Label, Wrapper } from '../styled/RegisterForm';

export type FormProps = {
  title: string;
  handleClick: (email: string, password: string) => void;
};

const FormRegister = ({ title, handleClick }: FormProps) => {
  const navigate = useNavigate();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  console.log(email, password);

  return (
    <Body style={{ backgroundColor: 'black' }}>
      <Wrapper>
        <div className="shape"></div>
        <div className="shape"></div>
      </Wrapper>
      <Form>
        <h3>Login Here</h3>
        <Label htmlFor="username">Username</Label>
        <Input
          type="text"
          placeholder="Email or Phone"
          id="username"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
        <Label htmlFor="password">Password</Label>
        <Input
          type="password"
          placeholder="Password"
          id="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <Button onClick={() => handleClick(email, password)}>{title}</Button>
        <Button onClick={() => navigate('/register')}>Registration</Button>
      </Form>
    </Body>
  );
};

export default FormRegister;
