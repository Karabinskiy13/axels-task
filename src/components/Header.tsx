import React from 'react';

import { useNavigate } from 'react-router-dom';
import { AppBar, Toolbar, Typography } from '@mui/material';
import CameraAltIcon from '@mui/icons-material/CameraAlt';
import BarChartIcon from '@mui/icons-material/BarChart';

import { LinkTo } from '../styled/PictureList';

export const Header = () => {
  const navigate = useNavigate();
  return (
    <div>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h5"
            component="div"
            sx={{ flexGrow: 1 }}
            display="flex"
            alignItems="center"
            justifyContent="center"
            marginLeft="64px">
            <CameraAltIcon />
            Picture Application
          </Typography>
          <LinkTo role="button" onClick={() => navigate('/charts')} style={{ color: 'white' }}>
            <BarChartIcon />
          </LinkTo>
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Header;
