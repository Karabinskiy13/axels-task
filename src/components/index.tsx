export { default as ModalView } from './ModalView';
export { default as PictureList } from './PictureList';
export { default as SinglePicture } from './SinglePicture';
export { default as Chart } from './Chart';
export { default as Header } from './Header';
