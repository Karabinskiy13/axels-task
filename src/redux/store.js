import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import pictureReducer from '../redux/ducks/pictures';

const rootReducer = combineReducers({ pictureReducer });

const persistConfig = {
  key: 'root',
  storage
};
const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
      }
    })
});
export const persistor = persistStore(store);

export default store;
