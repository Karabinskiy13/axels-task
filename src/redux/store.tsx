import { configureStore, combineReducers } from '@reduxjs/toolkit';

import pictureReducer from './ducks/pictures';

const rootReducer = combineReducers({ pictureReducer });

const store = configureStore({
  reducer: rootReducer
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;
