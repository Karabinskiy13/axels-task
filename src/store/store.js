import { configureStore, combineReducers } from '@reduxjs/toolkit';

import pictureReducer from './slice/picture.slice';

const rootReducer = combineReducers({ pictureReducer });

const store = configureStore({
  reducer: rootReducer
});

export default store;
