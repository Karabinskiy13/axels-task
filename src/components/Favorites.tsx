import React from 'react';

import { useNavigate } from 'react-router';
import { useSelector, useDispatch } from 'react-redux';

import { RootState } from '../redux/store';
import SinglePicture from './SinglePicture';
import { removeFromFavorites } from '../redux/ducks/pictures';

import { Grid, Container, Button, Typography, AppBar, Toolbar } from '@mui/material';

const Favorites = () => {
  const { favorites } = useSelector((state: RootState) => state.pictureReducer);
  const dispatch = useDispatch();
  const navigate = useNavigate();

  return (
    <div>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h5"
            component="div"
            sx={{ flexGrow: 1 }}
            display="flex"
            alignItems="center"
            justifyContent="center"
            marginLeft="64px">
            Favorites
          </Typography>
        </Toolbar>
      </AppBar>
      <Button style={{ margin: '10px' }} onClick={() => navigate('/')} variant="outlined">
        Back to Pictures
      </Button>
      <Container maxWidth="xl" style={{ marginTop: '30px' }}>
        <Grid container spacing={2}>
          {favorites &&
            favorites.map((picture) => (
              <SinglePicture
                key={picture.previewURL}
                picture={picture}
                showModal={() => null}
                remove={() => dispatch(removeFromFavorites(picture.previewURL))}
              />
            ))}
        </Grid>
      </Container>
    </div>
  );
};

export default Favorites;
