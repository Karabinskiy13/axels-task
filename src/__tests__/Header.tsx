import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { ReactRouter6Adapter } from 'use-query-params/adapters/react-router-6';
import { QueryParamProvider } from 'use-query-params';

import store from '../redux/store';
import { Header } from '../components';

test('Should render Header component', async () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <QueryParamProvider adapter={ReactRouter6Adapter}>
          <Header />
        </QueryParamProvider>
      </BrowserRouter>
    </Provider>
  );
  await screen.findByText('Picture Application');
  expect(screen.getByText('Picture Application')).toHaveTextContent('Picture Application');
  expect(asFragment()).toMatchSnapshot();
});
