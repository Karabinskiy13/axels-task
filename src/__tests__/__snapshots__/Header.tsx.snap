// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> Should render component 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-qjh99k-MuiTypography-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CameraAltIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="3.2"
            />
            <path
              d="M9 2 7.17 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2h-3.17L15 2H9zm3 15c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
            />
          </svg>
          Picture Application
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium sc-hKMtZM kdLpAF css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          role="button"
          style="color: white;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="BarChartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 9h4v11H4zm12 4h4v7h-4zm-6-9h4v16h-4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
